OSGi test failure due to missing environment varialble

OSGi test case `PropertyResolveConfigProviderOSGITest` fails due to missing environment variable `server.name` when building Carbon-Kernel 5.2.0-SNAPSHOT

The full build log is as shown below:

```
JAVA_HOME environment variable is set to /usr/lib/jvm/java-8-oracle
CARBON_HOME environment variable is set to /home/viduran/Documents/kernel/carbon-kernel/tests/osgi-tests/target/bd88bd8d-3ad9-4f1a-abc2-3fee6b94fc8c
RUNTIME_HOME environment variable is set to /home/viduran/Documents/kernel/carbon-kernel/tests/osgi-tests/target/bd88bd8d-3ad9-4f1a-abc2-3fee6b94fc8c
osgi> [2017-05-25 11:56:34,516] ERROR {org.wso2.carbon.config.provider.ConfigProviderImpl} - Environment variable server.name not found. Placeholder: id: test-carbon-kernel
name: ${env:server.name}
tenant: ${sec:wso2.sample.tenant}
ports:
  offset: 0
startupResolver:
  capabilityListenerTimer:
    delay: 200
    period: 200
  pendingCapabilityTimer:
    delay: 60000
    period: 30000
jmx:
  enabled: false
  hostName: 127.0.0.1
  rmiServerPort: 11111
  rmiRegistryPort: 9999

[2017-05-25 11:56:34,516] ERROR {org.wso2.carbon.config.provider.ConfigProviderImpl} - Environment variable server.name not found. Placeholder: id: test-carbon-kernel
name: ${env:server.name}
tenant: ${sec:wso2.sample.tenant}
ports:
  offset: 0
startupResolver:
  capabilityListenerTimer:
    delay: 200
    period: 200
  pendingCapabilityTimer:
    delay: 60000
    period: 30000
jmx:
  enabled: false
  hostName: 127.0.0.1
  rmiServerPort: 11111
  rmiRegistryPort: 9999

[2017-05-25 11:56:34,520] ERROR {org.wso2.carbon.kernel.internal.CarbonCoreComponent} - Error while activating CarbonCoreComponent
[org.ops4j.pax.exam.testng.listener.PaxExam] : Exception
java.lang.RuntimeException: org.ops4j.pax.exam.TestContainerException: java.lang.reflect.InvocationTargetException
	at org.ops4j.pax.exam.rbc.client.intern.RemoteBundleContextClientImpl$1.invoke(RemoteBundleContextClientImpl.java:105)
	at com.sun.proxy.$Proxy8.call(Unknown Source)
	at org.ops4j.pax.exam.rbc.client.intern.RemoteBundleContextClientImpl.call(RemoteBundleContextClientImpl.java:268)
	at org.ops4j.pax.exam.container.remote.RBCRemoteTarget.call(RBCRemoteTarget.java:60)
	at org.wso2.carbon.container.CarbonTestContainer.call(CarbonTestContainer.java:393)
	at org.ops4j.pax.exam.spi.reactors.EagerSingleStagedReactor.invoke(EagerSingleStagedReactor.java:109)
	at org.ops4j.pax.exam.testng.listener.PaxExam.runByDriver(PaxExam.java:458)
	at org.ops4j.pax.exam.testng.listener.PaxExam.run(PaxExam.java:308)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:212)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:652)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:295)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:84)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:90)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
Caused by: org.ops4j.pax.exam.TestContainerException: java.lang.reflect.InvocationTargetException
	at org.ops4j.pax.exam.raw.extender.intern.ProbeInvokerImpl.injectContextAndInvoke(ProbeInvokerImpl.java:133)
	at org.ops4j.pax.exam.raw.extender.intern.ProbeInvokerImpl.findAndInvoke(ProbeInvokerImpl.java:85)
	at org.ops4j.pax.exam.raw.extender.intern.ProbeInvokerImpl.call(ProbeInvokerImpl.java:73)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.ops4j.pax.exam.rbc.internal.RemoteBundleContextImpl.remoteCall(RemoteBundleContextImpl.java:80)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:346)
	at sun.rmi.transport.Transport$1.run(Transport.java:200)
	at sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:568)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:826)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:683)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:682)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.ops4j.pax.exam.raw.extender.intern.ProbeInvokerImpl.injectContextAndInvoke(ProbeInvokerImpl.java:125)
	... 24 more
Caused by: org.wso2.carbon.config.ConfigurationRuntimeException: Environment variable server.name not found. Placeholder: id: test-carbon-kernel
name: ${env:server.name}
tenant: ${sec:wso2.sample.tenant}
ports:
  offset: 0
startupResolver:
  capabilityListenerTimer:
    delay: 200
    period: 200
  pendingCapabilityTimer:
    delay: 60000
    period: 30000
jmx:
  enabled: false
  hostName: 127.0.0.1
  rmiServerPort: 11111
  rmiRegistryPort: 9999

	at org.wso2.carbon.config.provider.ConfigProviderImpl.processValue(ConfigProviderImpl.java:244)
	at org.wso2.carbon.config.provider.ConfigProviderImpl.processPlaceholder(ConfigProviderImpl.java:186)
	at org.wso2.carbon.config.provider.ConfigProviderImpl.getConfigurationObject(ConfigProviderImpl.java:102)
	at org.wso2.carbon.osgi.config.PropertyResolveConfigProviderOSGITest.testPropertyResolve(PropertyResolveConfigProviderOSGITest.java:66)
	... 29 more
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 36.38 sec <<< FAILURE! - in TestSuite
testPropertyResolve(org.wso2.carbon.osgi.config.PropertyResolveConfigProviderOSGITest)  Time elapsed: 15.227 sec  <<< FAILURE!
org.wso2.carbon.config.ConfigurationRuntimeException: Environment variable server.name not found. Placeholder: id: test-carbon-kernel
name: ${env:server.name}
tenant: ${sec:wso2.sample.tenant}
ports:
  offset: 0
startupResolver:
  capabilityListenerTimer:
    delay: 200
    period: 200
  pendingCapabilityTimer:
    delay: 60000
    period: 30000
jmx:
  enabled: false
  hostName: 127.0.0.1
  rmiServerPort: 11111
  rmiRegistryPort: 9999

	at org.wso2.carbon.config.provider.ConfigProviderImpl.processValue(ConfigProviderImpl.java:244)
	at org.wso2.carbon.config.provider.ConfigProviderImpl.processPlaceholder(ConfigProviderImpl.java:186)
	at org.wso2.carbon.config.provider.ConfigProviderImpl.getConfigurationObject(ConfigProviderImpl.java:102)
	at org.wso2.carbon.osgi.config.PropertyResolveConfigProviderOSGITest.testPropertyResolve(PropertyResolveConfigProviderOSGITest.java:66)


Results :

Failed tests: 
  PropertyResolveConfigProviderOSGITest.testPropertyResolve:66 » ConfigurationRuntime
```

--- a/tests/osgi-tests/src/test/java/org/wso2/carbon/osgi/config/PropertyResolveConfigProviderOSGITest.java
+++ b/tests/osgi-tests/src/test/java/org/wso2/carbon/osgi/config/PropertyResolveConfigProviderOSGITest.java
@@ -56,7 +56,7 @@ public class PropertyResolveConfigProviderOSGITest {
 
     @Configuration
     public Option[] createConfiguration() {
-        setEnvProperty("server.name", "Test WSO2 Carbon Kernel");
+        setEnvProperty("SERVER_NAME", "Test WSO2 Carbon Kernel");
         return new Option[]{ copyCarbonYAMLOption(), systemProperty("server.id").value("test-carbon-kernel"),
                 copySecretPropertiesOption() };
     }

--- a/tests/osgi-tests/src/test/resources/carbon-config/deployment.yaml
+++ b/tests/osgi-tests/src/test/resources/carbon-config/deployment.yaml
@@ -19,7 +19,7 @@ wso2.carbon:
     # value to uniquely identify a server
   id: ${sys:server.id}
     # server name
-  name: ${env:server.name}
+  name: ${env:SERVER_NAME}
   tenant: ${sec:wso2.sample.tenant}
 
     # ports used by this server

